### PR TITLE
Istio upgrade warning

### PR DESCRIPTION
**Pull request by brendarearden**
### Issue
https://github.com/rancher/rancher/issues/35140

### Problem
Istio does not support upgrades across more than two minor versions

### Solution
Add warning section to app-readme so that the warning is visible in the ui when users are installing or upgrading rancher-istio

### CODE REVIEW

**Comment by brendarearden** [comment]
I did not bump the chart version because 1.11.4 has not been released yet.

File: packages/rancher-istio/charts/app-readme.md [hunk 1]
```diff
@@ -5,10 +5,13 @@ Our [Istio](https://istio.io/) installer wraps the istioctl binary commands in a
 * **[Jaeger](https://www.jaegertracing.io/)**: A quick start, all-in-one installation used for tracing distributed system. This is not production qualified, please refer to jaeger documentation to determine which installation you may need instead.
 
 For more information on how to use the feature, refer to our [docs](https://rancher.com/docs/rancher/v2.x/en/istio/v2.5/).
+## Warnings
+- Upgrading across more than two minor versions (e.g., 1.6.x to 1.9.x) in one step is not officially tested or recommended. See [Istio upgarde docs](https://istio.io/latest/docs/setup/upgrade/ "Istio docs") for more details.
```
Comment (jiaqiluo) [inline packages/rancher-istio/charts/app-readme.md:9]:
```suggestion
- Upgrading across more than two minor versions (e.g., 1.6.x to 1.9.x) in one step is not officially tested or recommended. See [Istio upgrade docs](https://istio.io/latest/docs/setup/upgrade/) for more details.
```

File: packages/rancher-istio/charts/app-readme.md [hunk 2]
```diff
@@ -5,10 +5,13 @@ Our [Istio](https://istio.io/) installer wraps the istioctl binary commands in a
...
+
 ## Known Issues
 
 #### Airgapped Environments
-If you are using this chart in an airgapped environment, you will not be able to upgrade. This is because the `istioctl` upgrade command reaches out to an external repo and it is not configurable. We are tracking the fix for this issue [here](https://github.com/rancher/rancher/issues/33402)
+**A temporary fix has been added to this chart to allow upgrades to succeed in an airgapped environment. See [this issue](https://github.com/rancher/rancher/issues/30842 "this issue") for details.** We are still advocating for an upstream fix in Istio to formally resolve this issue.The root cause is the Istio Operator upgrade command reaches out to an external repo on upgrades and the external repo is not configurable. We are tracking the fix for this issue [here](https://github.com/rancher/rancher/issues/33402)
```
Comment (jiaqiluo) [inline packages/rancher-istio/charts/app-readme.md:14]:
```suggestion
**A temporary fix has been added to this chart to allow upgrades to succeed in an airgapped environment. See [this issue](https://github.com/rancher/rancher/issues/30842) for details.** We are still advocating for an upstream fix in Istio to formally resolve this issue. The root cause is the Istio Operator upgrade command reaches out to an external repo on upgrades and the external repo is not configurable. We are tracking the fix for this issue [here](https://github.com/rancher/rancher/issues/33402)
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: packages/rancher-istio/charts/app-readme.md (hunk 1 above)
Comment (brendarearden) [reply packages/rancher-istio/charts/app-readme.md]: Corrected spelling in recent commit.

File: packages/rancher-istio/charts/app-readme.md (hunk 2 above)
Comment (brendarearden) [reply packages/rancher-istio/charts/app-readme.md]: Corrected spacing issue in recent commit.

**Review by jiaqiluo — APPROVED** [review]

**Review by doflamingo721 — APPROVED** [review]
LGTM

**Pull request merged**